### PR TITLE
Add map colorbar and fix transient bug

### PR DESCRIPTION
**Pull request by joemoorhouse**
Added a colorbar to the map; colorbar info stored as json but coded within the library for now: this will be exposed via an API call in a future version. 

useMemo for updating maps was also causing a transient bug (basically a race condition), so rewrote this.

### CODE REVIEW

**Comment by negillett** [comment]
Do you have a screenshot handy of this change in effect?

**Comment by joemoorhouse** [comment]
image

Here's a screenshot

**Comment by joemoorhouse** [comment]
Shown with the box merged from the change of @deepanshu44...

File: src/components/ScatterMap.js [hunk 1]
```diff
@@ -230,21 +241,53 @@ export default function ScatterMap(props) {
...
+              margin={{ top: 7, right: 7, left: 7, bottom: 0 }} backgroundColor='white'>
+              <defs>
+                <linearGradient id={"colorUv"} x1="0" y1="0" x2="1" y2="0">
+                {colorbarStops.map((prop, key) => {
+                  return (
+                    <stop offset={prop.offset} stopColor={prop.stopColor} key={key} />
+                  );
+                })} 
+                  {/* <stop offset="0%" stopColor="#FF0000" stopOpacity={1.0}/>
+                  <stop offset="100%" stopColor="#FFFFFF" stopOpacity={1.0}/> */}
```
Comment (negillett) [inline src/components/ScatterMap.js:276]: Can this be cleaned up?

**Review by negillett — APPROVED** [review]

**Pull request merged**